### ISSUE TITLE
Crash in 1.9.0 analysis engine when rapidly starting and stopping

### ISSUE THREAD

**Issue by rooklift** [lightvector/KataGo#512]
As of KataGo 1.9.0, I seem to run into some sort of crash when I rapidly start and stop searches in the analysis engine.

I was able to crash it by having this exchange in a fraction of a second:

```
>   {"id":"node_2:1","rules":"Japanese","komi":5.5,"boardXSize":19,"boardYSize":19,"maxVisits":1000000,"reportDuringSearchEvery":0.1,"includeOwnership":true,"includeMovesOwnership":true,"overrideSettings":{"reportAnalysisWinratesAs":"SIDETOMOVE","wideRootNoise":0.05},"initialStones":[],"moves":[],"initialPlayer":"B"}
>   {"id":"stop!node_2:1","action":"terminate","terminateId":"node_2:1"}
<== {"id":"node_2:1","isDuringSearch":false,"noResults":true,"turnNumber":0}
>   {"id":"node_3:3","rules":"Japanese","komi":5.5,"boardXSize":19,"boardYSize":19,"maxVisits":1000000,"reportDuringSearchEvery":0.1,"includeOwnership":true,"includeMovesOwnership":true,"overrideSettings":{"reportAnalysisWinratesAs":"SIDETOMOVE","wideRootNoise":0.05},"initialStones":[],"moves":[["B","R16"]]}
<== {"action":"terminate","id":"stop!node_2:1","terminateId":"node_2:1"}
>   {"id":"stop!node_3:3","action":"terminate","terminateId":"node_3:3"}
<== {"action":"terminate","id":"stop!node_3:3","terminateId":"node_3:3"}
```

Then the engine apparently crashed. Note that the GUI never received a `{isDuringSearch: false}` message for the node_3:3 search.

**Comment by rooklift** [comment]
I've made a rather lame Node script that reproduces the crash for me:

https://github.com/rooklift/katago_190_crash_test

**Comment by lightvector** [comment]
Fixed in 1.9.1 https://github.com/lightvector/KataGo/releases
Thanks for reporting!
